### PR TITLE
Drawsprite && new style && th locale 

### PR DESCRIPTION
**Pull request by iwindd**
// add
-- new style
-- draw sprite on everything can target
-- thai locale

// fix
-- fix no close when selected

### CODE REVIEW

**Comment by thelindat** [comment]
Nobody asked for these stupid style changes. 
image
image

What's the point of drawing sprites (which you randomly again changed the style of) for entities that don't have options?
image

Do you really expect me to merge this? I already deal with complaints about performance when people leave debug on or register their stupid options inside while loops. 
image

File: client/api.lua [hunk 1]
```diff
@@ -271,6 +283,10 @@ function api.removeLocalEntity(arr, options)
     end
 end
 
+function api.getLocalEntities()
+    return localEntities
+end
+
```
Comment (thelindat) [inline client/api.lua:289]: Cool random exports to get a bunch of data nobody should be using.

File: client/debug.lua [hunk 2]
```diff
@@ -7,17 +7,18 @@ AddEventHandler('ox_target:debug', function(data)
 	print(json.encode(data, {indent=true}))
 end)
 
-if GetConvarInt('ox_target:debug', 0) ~= 1 then return end
+if GetConvarInt('ox_target:debug', 0) ~= 1 then return end
```
Comment (thelindat) [inline client/debug.lua:10]: Random space.

File: client/debug.lua [hunk 3]
```diff
@@ -7,17 +7,18 @@ AddEventHandler('ox_target:debug', function(data)
...
 local drawZones = false
 
 ox_target:addBoxZone({
     coords = vec3(442.5363, -1017.666, 28.85637),
-    size = vec3(2, 2, 2),
+    size = vec3(2, 2, 3),
     rotation = 45,
     debug = drawZones,
     drawSprite = true,
+    debug= true,
```
Comment (thelindat) [inline client/debug.lua:21]: Randomly add a second debug field to this option.

File: client/main.lua [hunk 4]
```diff
@@ -33,7 +33,19 @@ local DisableControlAction = DisableControlAction
 local DisablePlayerFiring = DisablePlayerFiring
 local options = {}
 local currentTarget = {}
+local activated = false
```
Comment (thelindat) [inline client/main.lua:36]: ????

File: client/main.lua [hunk 5]
```diff
@@ -33,7 +33,19 @@ local DisableControlAction = DisableControlAction
...
+    return currentTarget
+end
+
+function getOptionsData()
+    return options
+end
+
+function getActivated()
+    return activated
+end
```
Comment (thelindat) [inline client/main.lua:48]: I _just_ got rid of a bunch of global functions and now you want to add a bunch of dumb ones. Also - _activated_???????????

File: client/main.lua [hunk 6]
```diff
@@ -53,6 +66,7 @@ local function startTargeting()
         hit, entityHit, endCoords = raycastFromCamera(flag)
         distance = #(playerCoords - endCoords)
 
+
```
Comment (thelindat) [inline client/main.lua:69]: I do like lines.

File: client/main.lua [hunk 7]
```diff
@@ -86,13 +100,15 @@ local function startTargeting()
...
                     table.wipe(options)
+                    activated = false
                     SendNuiMessage('{"event": "leftTarget"}')
                 end
 
                 if flag ~= 511 then
                     entityHit = HasEntityClearLosToEntity(entityHit, cache.ped, 7) and entityHit or 0
                 end
 
+
```
Comment (thelindat) [inline client/main.lua:111]: Like I said, I like lines.

File: client/main.lua [hunk 8]
```diff
@@ -131,6 +147,7 @@ local function startTargeting()
             currentTarget.entity = entityHit
             currentTarget.coords = endCoords
             currentTarget.distance = distance
+            currentTarget.bone = nil
```
Comment (thelindat) [inline client/main.lua:150]: ?????????????????????????????????????

File: client/main.lua [hunk 9]
```diff
@@ -191,6 +208,7 @@ local function startTargeting()
                         end
                     end
 
+
```
Comment (thelindat) [inline client/main.lua:211]: Line.

File: client/main.lua [hunk 10]
```diff
@@ -355,6 +377,8 @@ RegisterNUICallback('select', function(data, cb)
     end
 
     if IsNuiFocused() then
+        activated = false
         state.setActive(false)
+        state.setNuiFocus(false)
```
Comment (thelindat) [inline client/main.lua:382]: ?????????

Still don't see what your stupid "activated" is for, and _how would somebody select if they don't have nui focus?_

File: client/utils.lua [hunk 11]
```diff
@@ -35,7 +37,7 @@ if GetConvarInt('ox_target:drawSprite', 1) == 1 then
     local SetDrawOrigin = SetDrawOrigin
     local DrawSprite = DrawSprite
     local ClearDrawOrigin = ClearDrawOrigin
-    local colour = vector(155, 155, 155, 175)
+    local colour = vector(255, 255, 255, 255)
```
Comment (thelindat) [inline client/utils.lua:40]: ?????

File: client/utils.lua [hunk 12]
```diff
@@ -64,10 +66,85 @@ if GetConvarInt('ox_target:drawSprite', 1) == 1 then
             end
         end
 
+
+
```
Comment (thelindat) [inline client/utils.lua:70]: Two lines for the price of one!

File: client/utils.lua [hunk 13]
```diff
@@ -64,10 +66,85 @@ if GetConvarInt('ox_target:drawSprite', 1) == 1 then
...
         return n > 0 and inRange, newZone
     end
 
+
+    function utils.getNearbyTarget(coords)
```
Comment (thelindat) [inline client/utils.lua:75]: Only used locally, yet in the module table.

File: client/utils.lua [hunk 14]
```diff
@@ -64,10 +66,85 @@ if GetConvarInt('ox_target:drawSprite', 1) == 1 then
...
+        coords = coords or GetEntityCoords(PlayerPedId())
```
Comment (thelindat) [inline client/utils.lua:76]: Coords is never defined, except we already have it elsewhere in the code and could easily call it.
We also don't use PlayerPedId anywhere in this resource, yet suddenly!

File: client/utils.lua [hunk 15]
```diff
@@ -64,10 +66,85 @@ if GetConvarInt('ox_target:drawSprite', 1) == 1 then
...
+        inRange = {}
+
+        local n = 0
+        local targetData  = getTargetData();
+        local activated   = getActivated();
+        local isActive    = state.isActive();
+
+        local getDistance = function(obj) 
+            return #(GetEntityCoords(PlayerPedId())-GetEntityCoords(obj))
```
Comment (thelindat) [inline client/utils.lua:85]: And now we call PlayerPedId and GetEntityCoords for every single entity known to the client, on every single frame.

File: client/utils.lua [hunk 16]
```diff
@@ -64,10 +66,85 @@ if GetConvarInt('ox_target:drawSprite', 1) == 1 then
...
+        end
+
+        local models = api.getModels();
+        for _, obj in pairs(GetGamePool("CObject")) do
+            local distance = getDistance(obj)
+            local model = GetEntityModel(obj);
+
+            if (DoesEntityExist(obj) and models[model] and (distance < 7)) then
```
Comment (thelindat) [inline client/utils.lua:93]: Oh, the object one checks if it's in models. Of course, vehicle and peds can also use models but don't; meanwhile this one never checks entities/localEntities.

File: client/utils.lua [hunk 17]
```diff
@@ -64,10 +66,85 @@ if GetConvarInt('ox_target:drawSprite', 1) == 1 then
...
+                n += 1;
+                inRange[n] = {
+                    drawSprite = data?.drawSprite or  true,
+                    colour = isHover and (data?.hover or hover) or (data?.colour or colour),
+                    coords = GetEntityCoords(obj)
+                }
+            end
+        end
+
+        local vehicles = api.getVehicle();
```
Comment (thelindat) [inline client/utils.lua:122]: Unused wow.

File: client/utils.lua [hunk 18]
```diff
@@ -64,10 +66,85 @@ if GetConvarInt('ox_target:drawSprite', 1) == 1 then
...
+            
+            if (DoesEntityExist(vehicle) and (distance < 7)) then
+                local isHover =  targetData?.entity == vehicle and activated 
+
+                if (isHover) then  
+                end
+
+                n += 1;
+                inRange[n] = {
+                    drawSprite = data?.drawSprite or  true,
```
Comment (thelindat) [inline client/utils.lua:134]: Data is never defined here.

File: client/utils.lua [hunk 19]
```diff
@@ -64,10 +66,85 @@ if GetConvarInt('ox_target:drawSprite', 1) == 1 then
...
+                    colour = isHover and (data?.hover or hover) or (data?.colour or colour),
+                    coords = GetEntityCoords(obj)
+                }
+            end
+        end
+
+        local vehicles = api.getVehicle();
+        for _, vehicle in pairs(GetGamePool("CVehicle")) do
```
Comment (thelindat) [inline client/utils.lua:123]: CVehicle! Draw sprite on all vehicle entities.

File: web/index.html [hunk 20]
```diff
@@ -1,13 +1,22 @@
...
+
+        <div id="finding">
+            <div class="circle circle1"></div>
+            <div class="circle circle2"></div>
+            <div class="circle circle3"></div>
+            <div class="circle circle4"></div>
+        </div>
+
+        <div id="options-wrapper"></div>
+        <script src="https://code.jquery.com/jquery-3.7.0.js" integrity="sha256-JlqSTELeR4TLqP0OG9dxM7yDPqX1ox/HfgiSLBj8+kM=" crossorigin="anonymous"></script>
```
Comment (thelindat) [inline web/index.html:19]: jquery!

File: web/js/main.js [hunk 21]
```diff
@@ -1,32 +1,43 @@
 import { createOptions } from './createOptions.js';
 
-const optionsWrapper = document.getElementById('options-wrapper');
-const body = document.body;
-const eye = document.getElementById('eyeSvg');
+const optionsWrapper = $('#options-wrapper');
```
Comment (thelindat) [inline web/js/main.js:3]: Ooooh rewriting vanilla JS to use jquery.

File: web/style.css [hunk 22]
```diff
@@ -1,80 +1,150 @@
...
+    0% {
+        transform: scale(0);
+        opacity: 1;
+    }
+
+    100% {
+        transform: scale(0.75);
+        opacity: 0;
+    }
+}
```
Comment (thelindat) [inline web/style.css:150]: Formatting changes are just the best.

**Comment by thelindat** [comment]
Next time save us all time and ask if this is even desired.

**Comment by iwindd** [comment]
I'm just trying to offer it. may not be very good But I didn't think you'd be so angry.

**Pull request closed without merging**